org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandlerTest.java

```diff
@@ -3621,7 +3624,8 @@ public void testCompletion_QualifiedName() throws Exception {
 				//@formatter:on
 		CompletionList list = requestCompletions(unit, "new Array");
 		assertFalse(list.getItems().isEmpty());
-		assertEquals("ArrayList<>()", list.getItems().get(0).getTextEdit().getLeft().getNewText());
+		CompletionItem resolved = server.resolveCompletionItem(list.getItems().get(0)).join();
+		assertEquals("ArrayList<>", resolved.getTextEdit().getLeft().getNewText());
```

Comment:
Why is this the new expected completion ?

Reply:
Now, the completion item is resolve lazily within `document/resolveCompletionItem`